bookwyrm/models/hashtag.py

```diff
@@ -0,0 +1,19 @@
+""" model for tags """
+from bookwyrm import activitypub
+from .activitypub_mixin import ActivitypubMixin
+from .base_model import BookWyrmModel
+from .fields import CICharField
+
+
+class Hashtag(ActivitypubMixin, BookWyrmModel):
+    "a hashtag which can be used in statuses"
+
+    name = CICharField(
```

Comment:
with the current regex, this will include the hash sign (i.e. `#rad`), I'm still a bit on the fence about this, but did it because eventually we might want to support other versions of the hash sign, i.e. with the Japanese keyboard a hash results in the [fullwidth number sign](https://unicode-table.com/en/FF03/). So being able to find the hashtag again inside a status might be much harder, plus this might be important for tools that might want to parse the content for the hashtag to replace the URL with their own?

Happy to change this around though and only store the value of the hashtag, without the signifier.

Reply:
What is the advantage to including the symbol? Your points against it seem compelling to me

Reply:
After looking at the federation piece a bit more I think it's better to keep the hash symbol as part of the tag. This way there's a canonical string that can be used to find the correct parts of the raw content. Also, other ActivityPub services include the symbol in the JSON representation where hashtags are split out as tags, i.e. here's an example from Mastodon ([this one](https://secretbearsociety.org/@chdorner/109857363838484273.json)):
<img width="458" alt="image" src="https://user-images.githubusercontent.com/112063/220204683-87f199a9-967a-4681-8cdb-72483c8c984e.png">

I have the feeling that it's better to keep the hashtag from the original message rather than adding the hard-coded symbol in the JSON representation. If for whatever reason we would want to support different symbols (i.e. the Japanese keyboard using a [different sign](https://unicode-table.com/en/FF03/)) in the future, we would still be able to find the location of the tag inside the content field.

Reply:
Sounds good!